libraries/Microsoft.Bot.Builder.Dialogs.Debugging/DebugAdapter.cs

```diff
@@ -145,11 +145,16 @@ async Task IMiddleware.OnTurnAsync(ITurnContext turnContext, NextDelegate next,
                 thread.Run.Post(Phase.Exited);
                 await UpdateThreadPhaseAsync(thread, null, cancellationToken).ConfigureAwait(false);
 
-                threadByContext.TryRemove(turnContext, out var ignored);
+                threadByContext.TryRemove(getThreadId(turnContext), out var ignored);
                 threads.Remove(thread);
             }
         }
 
+        private static string getThreadId(ITurnContext turnContext)
+        {
+            return $"{turnContext.Activity.ChannelId}-{turnContext.Activity.Id}";
```

Comment:
>turnContext.Activity [](start = 22, length = 20)

since this is middleware, is turnContext.Activity ever null?

Reply:
no, we always have an activity (even if it's artificial)